mev_inspect/classifiers/specs/zero_ex.py

```diff
@@ -225,10 +225,13 @@ def parse_swap(
 
 
 def _get_taker_token_in_amount(
-    taker_address: str, token_in_address: str, child_transfers: List[Transfer]
+    trace: DecodedCallTrace,
+    taker_address: str,
+    token_in_address: str,
+    child_transfers: List[Transfer],
 ) -> int:
 
-    if len(child_transfers) != 2:
+    if len(child_transfers) != 2 and trace.error == None:
```

Comment:
none checks are always `is None` or `is not None`

https://legacy.python.org/dev/peps/pep-0008/#programming-recommendations

Reply:
I think rather than having the “trickle through” down to that zero, it’d be nice to make it more explicit as its own check up front

```
if trace.error is not None:
    return 0
```

i don’t think we have any known cases besides this where that last zero should get hit right? in which case it should become an exception